Notebook for SPARCL + Jdaviz

weaverba137 opened the pull request:
This PR adds a how-to for displaying SPARCL spectra in Jdaviz (specifically Specivz).  This notebook was initially developed at [.Astronomy 12](https://www.dotastronomy.com/twelve/).

jacquesalice commented:
Hi @weaverba137 I've gone through and added section descriptions and other small formatting edits. Jdaviz is available on gp02 if you would like to review/test it on there. @rnikutta and @galaxyumi , can yall take a look at this notebook as well?

galaxyumi commented on 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:
```diff
@@ -0,0 +1,474 @@
+    "# SPARCL import\n",
+    "from sparcl.client import SparclClient\n",
+    "# Jdaviz import\n",
+    "from jdaviz import Specviz\n",
```
It seems that the module 'jdaviz' has not been installed by default for everyone. I am encountering the following error when running this cell: 'No module named 'jdaviz'.' Could you either make the package installation global or add instructions on how to install this module for individual accounts at the beginning of this NB?

weaverba137 replied:
It should now be installed on both `gp02` and `gp12`.

galaxyumi commented on 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:
```diff
@@ -0,0 +1,474 @@
+    "<a class=\"anchor\" id=\"searchsparcl\"></a>\n",
+    "## Search for spectra in SPARCL\n",
+    "Using the `client.find()` method, we obtain the following fields from the DESI EDR data set using SPARCL:  \n",
+    "- `sparcl_id` : Universally Unique Identifier for spectrum in SPARCL\n",
```
lower case 'u'?

weaverba137 replied:
"Universally Unique Identifier" is a genuine proper noun.

galaxyumi commented on 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:
```diff
@@ -0,0 +1,474 @@
+    "## Search for spectra in SPARCL\n",
+    "Using the `client.find()` method, we obtain the following fields from the DESI EDR data set using SPARCL:  \n",
+    "- `sparcl_id` : Universally Unique Identifier for spectrum in SPARCL\n",
+    "- `ra` : Right Ascension in degrees\n",
```
lower case 'a'?

weaverba137 replied:
"Right Ascension" is valid. Among other things, it's abbreviated `RA`, not `Ra`.

galaxyumi commented on 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:
```diff
@@ -0,0 +1,474 @@
+    "<a class=\"anchor\" id=\"retrievesparcl\"></a>\n",
+    "## Retrieve spectra from SPARCL and prepare plot data\n",
+    "Here we define a function that:\n",
+    "1. Uses the `client.retrieve()` method from SPARCL to retrieve spectra data for the list of IDs from DESI EDR.\n",
```
delete 'data'

weaverba137 replied:
Fixed.

galaxyumi reviewed (commented):
@weaverba137 and @jacquesalice Because the module 'jdaviz' was not installed for my account, my current review is very limited to text. I should be able to provide more complete feedback once you figure out a solution regarding the 'jdaviz' installation.

galaxyumi commented:
@weaverba137 and @jacquesalice, here are more comments:

1. The last two commente-out lines in the cell where you define the function `on_selected` can be deleted if they are no longer in use.
2. The last cell for making plot could have more description in the markdown. For example, featuring a drop down bar where one can select a target to display its spectrum with the best-fit model. 
3. The y-axis label is not following the scientific unit style conventions, e.g.,[erg s^{-1} cm^{-2} A^{-1}]. 
4. The x-axis ticks are too busy to read. Maybe changing to nm? I am getting a long warning box complaining about the Angstrom anyway. 
5. It would be great if the plot's legend could be more informative. Currently, they are random(?) two alphabets in red and black colors for model and data.  
6. Please take care of this long warning if possible. Otherwise, please leave a comment about the warnings in the markdown not to scare the users. <img width="737" alt="Screen Shot 2023-10-16 at 4 50 06 PM" src="https://github.com/astro-datalab/notebooks-latest/assets/5504376/8a6b218b-f2f2-46f0-9bb9-71346fe71a5c">

weaverba137 commented on 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:
```diff
@@ -0,0 +1,474 @@
+    "<a class=\"anchor\" id=\"intro\"></a>\n",
+    "## Introduction\n",
+    "\n",
+    "This notebook demonstrates how to find and retrieve spectroscopy data for certain objects from the DESI EDR data set using [SPARCL](https://astrosparcl.datalab.noirlab.eduhttps://astrosparcl.datalab.noirlab.edu) (SPectra Analysis and Retrievable Catalog Lab) and producing an interactive plot of an object's spectrum using [Jdaviz](https://jdaviz.readthedocs.io/en/latest/index.html).\n",
```
Fixed.

weaverba137 commented on 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:
```diff
@@ -0,0 +1,474 @@
+    "<a class=\"anchor\" id=\"intro\"></a>\n",
+    "## Introduction\n",
+    "\n",
+    "This notebook demonstrates how to find and retrieve spectroscopy data for certain objects from the DESI EDR data set using [SPARCL](https://astrosparcl.datalab.noirlab.eduhttps://astrosparcl.datalab.noirlab.edu) (SPectra Analysis and Retrievable Catalog Lab) and producing an interactive plot of an object's spectrum using [Jdaviz](https://jdaviz.readthedocs.io/en/latest/index.html).\n",
```
Fixed.

weaverba137 commented on 04_HowTos/SPARCL/Plot_SPARCL_Spectra_with_Jdaviz.ipynb:
```diff
@@ -0,0 +1,474 @@
+    "<a class=\"anchor\" id=\"intro\"></a>\n",
+    "## Introduction\n",
+    "\n",
+    "This notebook demonstrates how to find and retrieve spectroscopy data for certain objects from the DESI EDR data set using [SPARCL](https://astrosparcl.datalab.noirlab.eduhttps://astrosparcl.datalab.noirlab.edu) (SPectra Analysis and Retrievable Catalog Lab) and producing an interactive plot of an object's spectrum using [Jdaviz](https://jdaviz.readthedocs.io/en/latest/index.html).\n",
```
"display"

weaverba137 commented:
From the comments above:

1. Fixed.
2. Fixed.
3. Waiting on a fully viable notebook server to test. This may be an internal feature of Jdaviz that cannot be changed.
4. Waiting on a fully viable notebook server to test. This may be an internal feature of Jdaviz that cannot be changed.
5. The alphabetical indexes are an internal feature of Jdaviz. We cannot change that in this notebook.
6. Waiting on a fully viable notebook server to fix.

weaverba137 commented:
Updates:

* The warnings are now fixed.
* The y-axis label looks fine to me.
* The x-axis looks fine to me.

@jacquesalice, I think this is ready to go, but it should be tested on `gp02` when it becomes available again, and also tested on `gp12` when Jdaviz is installed on that system.

weaverba137 commented:
PS, I updated the HTML version on the last commit. Please double-check that it looks OK.

weaverba137 commented:
I've retested this notebook with the new 'Python 3' kernel on `gp02`. Please review again. There are no major changes from last time, but it has been a while since we were able to test this.

stephjuneau commented:
Hi @weaverba137 , @jacquesalice , @galaxyumi : checking on the status of this notebook. From Ben's last comment it was retested on gp02. What are the next steps after reviewing this version before it can be deployed to prod/gp12?

weaverba137 commented:
I believe it requires the Python 3.10 kernel to be installed on the production server, so that might also be a question for @mjfitzpatrick.

weaverba137 commented:
@jacquesalice, it should be possible to at least review this notebook on gp13 so that it is ready to merge if not yet fully merged.

jacquesalice reviewed (approved):
Notebook runs well on gp13. Approving this PR and awaiting merge until we switch over to gp13
